### ISSUE TITLE
Consider runtime dependency variations for cases such as graphics/drm-kmod

### ISSUE THREAD

**Issue by grahamperrin** [FreshPorts/freshports#207]
For example, https://www.freshports.org/graphics/drm-kmod/#requiredrun lists one runtime dependency: 

> 1. drm.ko : graphics/drm-fbsd12.0-kmod

The actual dependency varies according to the ABI e.g. 

```text
% date ; uname -v
Sun 31 Jan 2021 00:22:14 GMT
FreeBSD 14.0-CURRENT #84 main-53729367d3: Sat Jan 30 18:47:56 GMT 2021     root@mowa219-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
% pkg rquery '%do %R' drm-kmod
graphics/drm-current-kmod FreeBSD
graphics/drm-current-kmod poudriere
% 
```

Do we assume that readers will know to _not_ take the runtime dependency line too literally? 

(In the `drm-kmod` case, there's https://www.freshports.org/graphics/drm-kmod/#description a suitably explanatory description.)

----

Or is there some way of signalling that sometimes, a listed dependency may be not always true?

**Comment by dlangille** [comment]
Can you figure out a way to fake the ABI we are running on?

**Comment by grahamperrin** [comment]
… with a view to telling whether responses vary for multiple ABI-specific [pkg-rquery(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg-rquery(8)) queries for a single port?

**Comment by dlangille** [comment]
Yes. I think we have to first detect if there is any difference.  Perhaps we fake the current ABI. Most information is extracted via 'make -V' - and this is done in a `chroot`. Maybe that holds the solution.

**Comment by grahamperrin** [comment]
I also thought of `chroot`. 

My uneducated guess (before your comment) was a jail (?) for each ABI then within each jail, some use of `pkg rquery` (with `--no-repo-update`). 

If a `make` command can get what's required, this is surely preferable to querying a remote server ☑

**Comment by grahamperrin** [comment]
As a side note, I wonder whether `graphics/xpdf` will be a comparable case, given: 

* the mixture of `3.04_7` and `4.⋯` under https://www.freshports.org/graphics/xpdf/#packages
* `3.⋯` (all greater than `3.04_7`) under https://www.freshports.org/graphics/xpdf3/#packages
* some discussion of x and `xpdf3-3.04_11` not dependent upon Qt.

Prepare to hide this comment as off-topic. I may be confusing myself …
